Tools/scripts/pathfix.py

```diff
@@ -10,7 +10,17 @@
 # arguments).
 # The original file is kept as a back-up (with a "~" attached to its name),
 # -n flag can be used to disable this.
-#
+
+# Sometimes you may find shebangs with flags such as `#! /usr/bin/env python -si`.
+# Normally, pathfix overwrites the entire line, including the flags.
+# To keep flags from the original shebang line, use -f "".
```

Comment:
I would prefer to have one option to preserve existing flags and another to add more flags. Like:

```
pathfix.py --preserve-flags --add-flags "..." script
```

where `--add-flags "..."` would accept an arbitrary string like: `--add-flags "-X dev --check-hash-based-pycs=never"`.

I suggest to replace legacy optparse with newer argparse, it's easier to use "long flags" (like --preserve-flags) in argparse.

If you really want to short flags (single letter), I suggest:

* `--preserve-flags/-p`
* `--add-flags/-a FLAGS`

Reply:
[The Linux kernel treats everything following the first “word” in the shebang line as a single argument](https://lwn.net/Articles/630727/). Such flags as `-X dev --check-hash-based-pycs=never` are impossible here.